integration/spring-data/base/src/main/java/com/blazebit/persistence/spring/data/base/query/AbstractPartTreeBlazePersistenceQuery.java

```diff
@@ -372,6 +385,20 @@ private Sort getDynamicSort(Object[] values) {
             return parameters.potentiallySortsDynamically() ? new ParametersParameterAccessor(parameters, values).getSort()
                     : null;
         }
+
+        private <T> void applySpecification(CriteriaQuery<T> criteriaQuery, Object[] values) {
+            JpaParameter specificationParameter = parameters.getSpecificationParameter();
+            if (specificationParameter != null) {
+                int specificationParameterIndex = specificationParameter.getIndex();
+                Specification<T> specification = (Specification<T>) values[specificationParameterIndex];
+                Root<T> root = (Root<T>) criteriaQuery.getRoots().iterator().next();
+                CriteriaBuilder criteriaBuilder = getEntityManager().getCriteriaBuilder();
```

Comment:
Cast the criteriaQuery to `BlazeCriteriaQuery` instead and use `getCriteriaBuilder()` to access the JPA CriteriaBuilder object.